clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractHeartbeatRequestManager.java

```diff
@@ -182,7 +181,7 @@ public NetworkClientDelegate.PollResult poll(long currentTimeMs) {
         }
 
         // Case 1: The member is leaving
-        boolean heartbeatNow = membershipManager().state() == MemberState.LEAVING ||
+        boolean heartbeatNow = isLeavingGroup() ||
```

Comment:
why add the `#isLeavingGroup` method to the AbstractHeartbeatRequestManager instead of just calling `membershipManager#isLeavingGroup` here?

Basically I'm wondering why both the HeartbeatRequestManager and MembershipManager classes need to have this `#isLeavingGroup` method

Reply:
`membershipManager#isLeavingGroup` considers `LEAVING` and `PREPARE_LEAVING` to be leaving the group.
But in this flag, we only consider `LEAVING` since the flag determines whether the heartbeat should be sent immediately or not. `PREPARE_LEAVING` is the state that waits for callbacks to be executed.

The key point is that it's confusing that HeartbeatRequestManager and MembershipManager have a method with the same name. Clearly, HeartbeatRequestManager cares about the timing of a leave request, not the member state.
I will figure out a more appropriate name.